app/scripts/lib/config-loader.js

```diff
@@ -48,15 +60,57 @@ function (
         // did not receive the cookie.
       }
 
-      var self = this;
-      return xhr.getJSON('/config')
-          .then(function (config) {
-            self._config = config;
-            return config;
-          });
+      self._configPromise = self._xhr.getJSON('/config')
+        .fail(function (jqXHR) {
+          throw ConfigLoader.Errors.normalizeXHRError(jqXHR);
+        });
+
+      return self._configPromise;
     },
   };
 
+  var t = function (msg) {
+    return msg;
+  };
+
+  ConfigLoader.Errors = _.extend({}, Errors, {
+    ERRORS: {
+      SERVICE_UNAVAILABLE: {
+        errno: 998,
+        message: t('System unavailable, try again soon')
+      },
+      UNEXPECTED_ERROR: {
+        errno: 999,
+        message: t('Unexpected error')
+      }
+    },
+    NAMESPACE: 'config',
+
+    normalizeXHRError: function (xhr) {
+      var err;
+      var status = (xhr && xhr.status) || 0;
+      if (! status) {
+        err = this.toError('SERVICE_UNAVAILABLE');
+      } else {
+        // There are no expected errors when fetching config.
+        // All other errors are unexpected errors.
+        err = this.toError('UNEXPECTED_ERROR');
+      }
+
+      // Prepend the namespace and HTTP status to the message
+      // so at least *SOME* idea of the problem is given.
+      // The error will be reported to the error metrics.
+      err.message =
+        Strings.interpolate('%(namespace)s (%(status)s): %(message)s', {
```

Comment:
It seems like we should use tags then, here are the docs: https://raven-js.readthedocs.org/en/latest/usage.html#passing-additional-data

Benefit of tags is that we can also filter by tags and see the % when those happen.

Let me know if you want to update this to use tags instead. Otherwise the PR looks good